scripts/unit/circle-env-spec.js

```diff
@@ -1,112 +1,89 @@
-const fs = require('fs').promises
-const sinon = require('sinon')
 const { expect } = require('chai')
 const { _checkCanaries } = require('../circle-env')
 
 describe('circle-env', () => {
   let cachedEnv = { ...process.env }
 
   afterEach(() => {
-    sinon.restore()
-    Object.assign(process.env, cachedEnv)
+    process.env = cachedEnv
   })
 
   beforeEach(() => {
-    delete process.env.COPY_CIRCLE_ARTIFACTS
-    process.env.CI = 'true'
-    process.env.CIRCLE_INTERNAL_CONFIG = '/foo.json'
+    process.env = { CI: 'true' }
   })
 
   context('with missing canaries', () => {
-    it('fails', async () => {
-      sinon.stub(fs, 'readFile')
-      .withArgs('/foo.json').resolves(JSON.stringify({
-        Dispatched: { TaskInfo: { Environment: { somekey: 'someval' } } },
-      }))
-
-      try {
-        await _checkCanaries()
-        throw new Error('should not reach')
-      } catch (err) {
-        expect(err.message).to.include('Missing MAIN_CANARY')
-      }
-    })
-
-    context('with no circleEnv', () => {
-      beforeEach(() => {
-        sinon.stub(fs, 'readFile')
-        .withArgs('/foo.json').resolves(JSON.stringify({
-          Dispatched: { TaskInfo: { Environment: {} } },
-        }))
+    context('internal PR', () => {
+      it('fails when neither canary is set', () => {
+        try {
+          console.log('before', process.env.MAIN_CANARY)
+          _checkCanaries()
```

Comment:
more of a nit, but removing the error below might make this test pass silently as a side effect, which makes me think we might want to opt to something like this?
```suggestion
                  expect(_checkCanaries()).to.throw('Missing MAIN_CANARY')

```

Reply:
updated [82b333b](https://github.com/cypress-io/cypress/pull/31264/commits/82b333b838eae2905e1783aa34f59fa7a66caae2)